### DISCUSSION TITLE
AAPS treatment -> xDrip sync config

### DISCUSSION THREAD

**Discussion by Amidamaru** [NightscoutFoundation/xDrip#3247] [Q&A] [answered]
Hey guys,
I struggle to get all my treatments from AAPS to xDrip. Until now I was using Nightscout (at fly.io) for exchanging the bgs, treatments etc between AAPS and xDrip. I wanted to get rid of this and do it via local broadcast. But I struggle to set it up correctly.
I searched the documentations and different discussions (e.g. https://github.com/NightscoutFoundation/xDrip/discussions/2636) but even then it is not working.

My setup:
- xDrip Nightly 2123120516
- Libre 3
- Juggluco (enabled xdrip broadcast and patched broadcast)
- AndroidAPS 3.2.0.2 (getting BGs directly from Juggluco now)
- *AAPS: enabled broadcast to xDrip+ 
*I also saw the hint here to enable "xDrip+ Sync Follower" as main source in xDrip. This is fine for me, but what is the master in this way? I tried localhost or 127.0.0.1 , but no luck. Nothing is received. 

In xDrip I have enabled "local broadcast"+ "Accept Glucose" + "Accept treatments". I also played a bit with different settings for "identify receiver" ...
Can anyone support me to set this correctly? I really tried lots of configurations...
Any ideas?

My goal is pretty simple: I want to get rid of the necessity to have nightscout in fly.io to exchange all treatments between AAPS and XDrip, especially when both apps are running on my mobilephone :)

**Comment by PhilTU** [comment]
xDrip+ sync follower as far as I know uses Google Play Services and is for one xDrip+ to follow another ? https://navid200.github.io/xDrip/docs/xDripSync.html or https://xdrip.readthedocs.io/en/latest/use/syncsetup/

**Reply by Amidamaru** [reply]
heyho! ye exactly I also found these documentation and this also raised my confusion when reading the note in AAPS app config: _Send glucose and treatments data to xDrip+. Data Source \"xDrip+ Sync Follower\" must be selected and accepting of data must be enabled in Settings - Inter-app settings - Accept Glucose/Treatments_

**Reply by PhilTU** [reply]
Is that in AAPS Docs or in line help on the app. I didn't see that reference.

xDrip sync is intended for remote following and I'm not familiar with using it on local host.

You can turn off the patched app option in Juggluco if xDrip sync follower is being used. It'll only transfer glucose data if Libre 2 Patched is the Hardware Data Source

**Comment by Navid200** [comment]
Have you also reached out to the AAPS team or an AAPS forum to see if there may be any suggestions?

**Reply by Amidamaru** [reply]
hi! not yet, because I was not sure if xDrip or AAPS might help me better :D

**Comment by Amidamaru** [comment, marked as answer]
I fixed it finally :)

Today I realized in xDrip I see all the treatments like intended. I switched off the setting "Local Broadcast" in xDrip and I think this did the trick.

so my final working setup:
- xDrip Nightly 2123120516 
1.  disable local broadcast
2. enable "accept treatment" and "accept glucose"
3. Data source. Libre patched
- Libre 3
- Juggluco (enabled xdrip broadcast and patched broadcast)
- AndroidAPS 3.2.0.2 (getting BGs directly from Juggluco now)
